Core/Layer/Worlds/StatusBar/StatusBarRenderer.cs

```diff
@@ -0,0 +1,1160 @@
+    {
+        if (!StatusBarConditionResolver.Evaluate(context, anim.Conditions))
+            return;
+
+        Vec2I currentPos = ResolvePosition(anim, parentPos, widescreenOffset);
+
+        if (anim.Frames.Count > 0)
+        {
+            double totalDuration = anim.Frames.Sum(f => f.Duration);
+            if (totalDuration > 0)
+            {
+                double timePerLoop = totalDuration * Constants.TicksPerSecond;
+                long currentTick = WorldStatic.World.LevelTime;
+                double animTime = currentTick % timePerLoop;
+
+                string patch = anim.Frames[0].Lump;
+                double timeAccumulator = 0;
+
+                foreach (var frame in anim.Frames)
+                {
+                    timeAccumulator += frame.Duration * Constants.TicksPerSecond;
+                    if (animTime < timeAccumulator)
+                    {
+                        patch = frame.Lump;
+                        break;
+                    }
+                }
+
+                Align align = ConvertAlignment(anim.Alignment);
+                DrawSBarTexture(hud, patch, currentPos, align, useDoomOffsets: true, 
+                    translation: anim.Translation, alpha: 1.0f);
+            }
+        }
+
+        DrawChildren(hud, anim, currentPos, containerHeight, context, widescreenOffset);
+    }
+
+    private void DrawComponent(IHudRenderContext hud, StatusBarComponentDef comp, Vec2I parentPos, int containerHeight, StatusBarContext context, float widescreenOffset)
+    {
+        if (!StatusBarConditionResolver.Evaluate(context, comp.Conditions))
+            return;
+
+        Vec2I pos = ResolvePosition(comp, parentPos, widescreenOffset);
+        string text = string.Empty;
+        var config = WorldStatic.World.Config.Hud;
+        
+        float alpha = comp.Translucency ? 0.5f : 1.0f;
+        StatusBarAlignment alignment = comp.Alignment;
+
+        var sbarDef = WorldStatic.World.ArchiveCollection.Definitions.StatusBarDefinition;
+        var fontDef = sbarDef.HudFonts.FirstOrDefault(f => f.Name.EqualsIgnoreCase(comp.Font));
+        
+        int fontHeight = 8;
+        if (fontDef != null && StemToHelionFontMap.TryGetValue(fontDef.Stem, out var helionFontName))
+        {
+            int h = hud.GetFontMaxHeight(helionFontName);
+            if (h > 0) fontHeight = h;
+        }
+
+        switch (comp.Type.ToLowerInvariant())
+        {
+            case "stat_totals":
+                if (!config.ShowStats.Value) return;
+                DrawStatTotals(hud, comp, pos, fontDef, fontHeight, alpha);
+                DrawChildren(hud, comp, pos, containerHeight, context, widescreenOffset);
+                return;
+
+            case "time":
+                TimeSpan t = TimeSpan.FromSeconds(WorldStatic.World.LevelTime / 35.0);
+                text = $"{(int)t.TotalHours:D2}:{t.Minutes:D2}:{t.Seconds:D2}";
```

Comment:
Make text ReadOnlySpan<char> and use a SpanString as a member variable. This example comes from WorldLayer.Render.Hud DrawStatInfo:

                TimeSpan ts = TimeSpan.FromSeconds(World.LevelTime / 35);
                m_timeString.Clear();
                m_timeString.Append(ts.Hours, 2);
                m_timeString.Append(':');
                m_timeString.Append(ts.Minutes, 2);
                m_timeString.Append(':');
                m_timeString.Append(ts.Seconds, 2);